quiche/src/lib.rs

```diff
@@ -1408,6 +1408,12 @@ pub struct Connection {
     /// Total number of packets sent with data retransmitted.
     retrans_count: usize,
 
+    /// Total number of sent DATAGRAM frames.
+    dgram_sent_count: usize,
+
+    /// Total number of received DATAGRAM frames.
+    dgram_recv_count: usize,
```

Comment:
FWIW, in theory its possible for a minimally encoded length-prefixed DATAGRAM to take about 2 bytes and therefore a single packet to contain a lot of them. For a standard MTU, this would be like 600-700 but larger MTUs are possible. Therefore, there's a risk that the limit here could be breached orders of magnitude before the QUIC connection hit its packet count limit. 

Not sure if that warrants a saturating_add on the counting but wanted to flag before I forgot.

Reply:
Good point. I changed `+= 1` to `saturating_add(1)` and rebased the commit on top of master.